[Bug]: Kerberos authenticate will fail after a period of time

### What happened?

AMS and Optimizer will kerberos authenticate fail after a period of time,
and the task did not fail or restart.


### Affects Versions

master

### What engines are you seeing the problem on?

AMS, Optimizer

### How to reproduce

_No response_

### Relevant log output

```shell
2023-02-21 19:50:24,210 WARN  com.netease.arctic.optimizer.flink.FlinkConsumer             [] - failed to poll task, retry 0
org.apache.flink.streaming.runtime.tasks.ExceptionInChainedOperatorException: Could not forward element to next operator
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:99) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:56) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:29) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$ManualWatermarkContext.processAndCollect(StreamSourceContexts.java:418) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$WatermarkContext.collect(StreamSourceContexts.java:513) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$SwitchingOnClose.collect(StreamSourceContexts.java:103) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at com.netease.arctic.optimizer.flink.FlinkConsumer.run(FlinkConsumer.java:55) [arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110) [flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:67) [flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:323) [flink-dist_2.12-1.14.6.jar:1.14.6]
Caused by: java.lang.IllegalStateException: failed when load catalog hive_prod
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:151) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.loadCatalog(CatalogLoader.java:177) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:69) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:79) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.buildTable(BaseTaskExecutor.java:87) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.execute(BaseTaskExecutor.java:117) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:74) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:40) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.flink.streaming.api.operators.ProcessOperator.processElement(ProcessOperator.java:66) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	... 11 more
Caused by: java.lang.RuntimeException: Re-login from keytab failed
	at com.netease.arctic.table.TableMetaStore.getUGI(TableMetaStore.java:293) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:343) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.BasicIcebergCatalog.initialize(BasicIcebergCatalog.java:75) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:146) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.loadCatalog(CatalogLoader.java:177) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:69) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:79) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.buildTable(BaseTaskExecutor.java:87) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.execute(BaseTaskExecutor.java:117) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:74) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:40) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.flink.streaming.api.operators.ProcessOperator.processElement(ProcessOperator.java:66) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	... 11 more
Caused by: org.apache.hadoop.security.KerberosAuthException: Login failure for principal: /data/yarn/nm/usercache/hive/appcache/application_1667984545171_3215770/container_e38_1667984545171_3215770_01_000010/arctic_krb_conf/bb5c4f4f59f6ef3f166b556d40d4339c_300688@hd-059.com/krb.keytab from keytab /data/yarn/nm/usercache/hive/appcache/application_1667984545171_3215770/container_e38_1667984545171_3215770_01_000010/krb5.keytab javax.security.auth.login.LoginException: java.lang.IllegalArgumentException: Empty nameString not allowed
	at sun.security.krb5.PrincipalName.validateNameStrings(PrincipalName.java:174)
	at sun.security.krb5.PrincipalName.<init>(PrincipalName.java:397)
	at sun.security.krb5.PrincipalName.<init>(PrincipalName.java:468)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:650)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.apache.hadoop.security.UserGroupInformation.reloginFromKeytab(UserGroupInformation.java:1261)
	at org.apache.hadoop.security.UserGroupInformation.checkTGTAndReloginFromKeytab(UserGroupInformation.java:1168)
	at com.netease.arctic.table.TableMetaStore.getUGI(TableMetaStore.java:285)
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:343)
	at com.netease.arctic.catalog.BasicIcebergCatalog.initialize(BasicIcebergCatalog.java:75)
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:146)
	at com.netease.arctic.catalog.CatalogLoader.loadCatalog(CatalogLoader.java:177)
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:69)
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:79)
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.buildTable(BaseTaskExecutor.java:87)
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.execute(BaseTaskExecutor.java:117)
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:74)
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:40)
	at org.apache.flink.streaming.api.operators.ProcessOperator.processElement(ProcessOperator.java:66)
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82)
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57)
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29)
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:56)
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:29)
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$ManualWatermarkContext.processAndCollect(StreamSourceContexts.java:418)
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$WatermarkContext.collect(StreamSourceContexts.java:513)
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$SwitchingOnClose.collect(StreamSourceContexts.java:103)
	at com.netease.arctic.optimizer.flink.FlinkConsumer.run(FlinkConsumer.java:55)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:67)
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:323)

	at org.apache.hadoop.security.UserGroupInformation.reloginFromKeytab(UserGroupInformation.java:1270) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.checkTGTAndReloginFromKeytab(UserGroupInformation.java:1168) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.getUGI(TableMetaStore.java:285) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:343) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.BasicIcebergCatalog.initialize(BasicIcebergCatalog.java:75) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:146) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.loadCatalog(CatalogLoader.java:177) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:69) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:79) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.buildTable(BaseTaskExecutor.java:87) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.execute(BaseTaskExecutor.java:117) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:74) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:40) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.flink.streaming.api.operators.ProcessOperator.processElement(ProcessOperator.java:66) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	... 11 more
Caused by: javax.security.auth.login.LoginException: java.lang.IllegalArgumentException: Empty nameString not allowed
	at sun.security.krb5.PrincipalName.validateNameStrings(PrincipalName.java:174)
	at sun.security.krb5.PrincipalName.<init>(PrincipalName.java:397)
	at sun.security.krb5.PrincipalName.<init>(PrincipalName.java:468)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:650)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.apache.hadoop.security.UserGroupInformation.reloginFromKeytab(UserGroupInformation.java:1261)
	at org.apache.hadoop.security.UserGroupInformation.checkTGTAndReloginFromKeytab(UserGroupInformation.java:1168)
	at com.netease.arctic.table.TableMetaStore.getUGI(TableMetaStore.java:285)
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:343)
	at com.netease.arctic.catalog.BasicIcebergCatalog.initialize(BasicIcebergCatalog.java:75)
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:146)
	at com.netease.arctic.catalog.CatalogLoader.loadCatalog(CatalogLoader.java:177)
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:69)
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:79)
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.buildTable(BaseTaskExecutor.java:87)
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.execute(BaseTaskExecutor.java:117)
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:74)
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:40)
	at org.apache.flink.streaming.api.operators.ProcessOperator.processElement(ProcessOperator.java:66)
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82)
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57)
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29)
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:56)
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:29)
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$ManualWatermarkContext.processAndCollect(StreamSourceContexts.java:418)
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$WatermarkContext.collect(StreamSourceContexts.java:513)
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$SwitchingOnClose.collect(StreamSourceContexts.java:103)
	at com.netease.arctic.optimizer.flink.FlinkConsumer.run(FlinkConsumer.java:55)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:67)
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:323)

	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:856) ~[?:1.8.0_141]
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195) ~[?:1.8.0_141]
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682) ~[?:1.8.0_141]
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680) ~[?:1.8.0_141]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_141]
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680) ~[?:1.8.0_141]
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587) ~[?:1.8.0_141]
	at org.apache.hadoop.security.UserGroupInformation.reloginFromKeytab(UserGroupInformation.java:1261) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.hadoop.security.UserGroupInformation.checkTGTAndReloginFromKeytab(UserGroupInformation.java:1168) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.getUGI(TableMetaStore.java:285) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:343) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.BasicIcebergCatalog.initialize(BasicIcebergCatalog.java:75) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:146) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.loadCatalog(CatalogLoader.java:177) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:69) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.catalog.CatalogLoader.load(CatalogLoader.java:79) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.buildTable(BaseTaskExecutor.java:87) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.operator.BaseTaskExecutor.execute(BaseTaskExecutor.java:117) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:74) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at com.netease.arctic.optimizer.flink.FlinkExecuteFunction.processElement(FlinkExecuteFunction.java:40) ~[arctic-optimizer-0.5.0-SNAPSHOT-jar-with-dependencies-for-relogin-test.jar:?]
	at org.apache.flink.streaming.api.operators.ProcessOperator.processElement(ProcessOperator.java:66) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist_2.12-1.14.6.jar:1.14.6]
	... 11 more
```
```


### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -283,7 +283,7 @@ public class TableMetaStore implements Serializable {
                 }
               }
 
-              if (oldSystemPrincipal != null && oldSystemPrincipal.equals(krbPrincipal)) {
+              if (oldSystemPrincipal != null && !oldSystemPrincipal.equals(krbPrincipal)) {
                 System.setProperty("sun.security.krb5.principal", krbPrincipal);
                 systemPrincipalChanged = true;
               }

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -260,6 +260,8 @@ public class TableMetaStore implements Serializable {
               }
             }
 
+            String oldSystemPrincipal = System.getProperty("sun.security.krb5.principal");
+            boolean systemPrincipalChanged = false;
             try {
               if (!UserGroupInformation.isSecurityEnabled()) {
                 UserGroupInformation.setConfiguration(getConfiguration());
@@ -280,7 +282,11 @@ public class TableMetaStore implements Serializable {
                   LOG.warn("Fail to reflect UserGroupInformation", e);
                 }
               }
-              System.setProperty("sun.security.krb5.principal", krbPrincipal);
+
+              if (!krbPrincipal.equals(oldSystemPrincipal)) {
+                System.setProperty("sun.security.krb5.principal", krbPrincipal);
+                systemPrincipalChanged = true;
+              }
               ugi.checkTGTAndReloginFromKeytab();
             } catch (Exception e) {
               throw new RuntimeException("Re-login from keytab failed", e);
@@ -293,6 +299,9 @@ public class TableMetaStore implements Serializable {
                   UGI_REFLECT = false;
                   LOG.warn("Fail to reflect UserGroupInformation", e);
                 }
+              }
+              if (systemPrincipalChanged) {
+                System.setProperty("sun.security.krb5.principal", oldSystemPrincipal);
               }
             }
           }

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -280,6 +280,7 @@ public class TableMetaStore implements Serializable {
                   LOG.warn("Fail to reflect UserGroupInformation", e);
                 }
               }
+              System.setProperty("sun.security.krb5.principal", krbPrincipal);
               ugi.checkTGTAndReloginFromKeytab();
             } catch (Exception e) {
               throw new RuntimeException("Re-login from keytab failed", e);

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -283,7 +283,7 @@ public class TableMetaStore implements Serializable {
                 }
               }
 
-              if (!krbPrincipal.equals(oldSystemPrincipal)) {
+              if (oldSystemPrincipal != null && oldSystemPrincipal.equals(krbPrincipal)) {
                 System.setProperty("sun.security.krb5.principal", krbPrincipal);
                 systemPrincipalChanged = true;
               }

